src/cache/redis.rs

```diff
@@ -0,0 +1,32 @@
+use crate::twin::Twin;
+
+use super::Cache;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use bb8_redis::{bb8::Pool, RedisConnectionManager};
+
+//
+// how_to_init
+// let manager = RedisConnectionManager::new(url)?;
+// let pool = Pool::builder().build(manager).await?;
+//
+pub struct RedisCache {
+    pool: Pool<RedisConnectionManager>,
+}
+
+impl RedisCache {
+    pub async fn new(pool: Pool<RedisConnectionManager>) -> Result<Self> {
+        Ok(Self { pool })
+    }
+}
+
+#[async_trait]
+impl Cache<Twin> for RedisCache {
```

Comment:
The Redis Cache implementation can doesn't have to be against the concrete type Twin. This can still be generic type T.

The problem is to be able to store the type T inside redis, you will need to serialize it. So the solution for this is that the type T must implement serde `Serialize` and `Deserialize` traits. Which means the redis cache can work for any type that is serializable.

```suggestion
impl Cache<T> for RedisCache 
where T: Serialize + Deserialize {
```